lib/src/parser/recursive.dart

```diff
@@ -187,6 +187,41 @@ class RecursiveAstParser {
         );
       }
     }
+
+    // At this point, it is a TextAttribute, but handle cases
+    // with 'on-' and 'bind-' prefix.
+    if (decoratorToken.lexeme.startsWith('on-')) {
```

Comment:
Hmm, maybe `on-` and `bind-` could be tokens. Would that improve or harm the code flow do you think?